packages/keyring-api/src/api/transaction.ts

```diff
@@ -0,0 +1,275 @@
+import { StringNumberStruct, UuidStruct } from '../utils';
+
+/**
+ * This struct represents an asset.
+ *
+ * @example
+ * ```ts
+ * asset: {
+ *   id: 'eip155:1/slip44:60',
+ *   unit: 'ETH',
+ * },
+ * ```
+ */
+const AssetStruct = union([
+  object({
+    /**
+     * It is a fungible asset.
+     */
+    fungible: literal(true),
+
+    /**
+     * Asset type (CAIP-19).
+     */
+    type: CaipAssetTypeStruct,
+
+    /**
+     * Unit of the asset. This has to be one of the supported units for the
+     * asset, as defined by MetaMask.
+     */
+    unit: string(),
+  }),
+  object({
+    /**
+     * It is a non-fungible asset.
+     */
+    fungible: literal(false),
+
+    /**
+     * Asset ID (CAIP-19).
+     */
+    id: CaipAssetIdStruct,
+  }),
+]);
+
+/**
+ * This struct represents an amount of an asset.
+ *
+ * @example
+ * ```ts
+ * fee: {
+ *   amount: '0.01',
+ *   asset: {
+ *     id: 'eip155:1/slip44:60',
+ *     unit: 'ETH',
+ *   },
+ * },
+ * ```
+ */
+const AmountStruct = object({
+  /**
+   * Amount in decimal string format.
+   */
+  amount: StringNumberStruct,
+
+  /**
+   * Asset information.
+   */
+  asset: AssetStruct,
+});
+
+/**
+ * This struct represents a participant in a transaction.
+ *
+ * @example
+ * ```ts
+ * from: [
+ *   {
+ *     address: '0x1234...',
+ *     amount: '0.01',
+ *     asset: {
+ *       id: 'eip155:1/slip44:60',
+ *       unit: 'ETH',
+ *     },
+ *   },
+ * ],
+ * ```
+ */
+const ParticipantStruct = object({
+  /**
+   * Amount transferred from or to the participant.
+   */
+  ...AmountStruct.schema,
+
+  /**
+   * Participant address.
+   */
+  address: string(),
+});
+
+/**
+ * Transaction statuses.
+ */
+export enum TransactionStatus {
+  Submitted = 'submitted',
+  Pending = 'pending',
+  Confirmed = 'confirmed',
+  Failed = 'failed',
+}
+
+/**
+ * Transaction types.
+ */
+export enum TransactionType {
+  Send = 'send',
+  Receive = 'receive',
+}
+
+/**
+ * This struct represents a blockchain transaction.
+ *
+ * @example
+ * ```ts
+ * {
+ *   "id": "f5d8ee39a430901c91a5917b9f2dc19d6d1a0e9cea205b009ca73dd04470b9a6",
+ *   "chain": "bip122:000000000019d6689c085ae165831e93",
+ *   "account": "b9beb861-9761-4b97-89ce-d992be5f34da",
+ *   "status": "confirmed",
+ *   "timestamp": 1716367781,
+ *   "type": "send",
+ *   "from": [
+ *     {
+ *       "address": "bc1qrp0yzgkf8rawkuvdlhnjfj2fnjwm0m8727kgah",
+ *       "amount": "0.2001",
+ *       "asset": {
+ *         "fungible": true,
+ *         "type": "bip122:000000000019d6689c085ae165831e93/slip44:0",
+ *         "unit": "BTC"
+ *       }
+ *     }
+ *   ],
+ *   "to": [
+ *     {
+ *       "address": "bc1qrp0yzgkf8rawkuvdlhnjfj2fnjwm0m8727kgah",
+ *       "amount": "0.1",
+ *       "asset": {
+ *         "fungible": true,
+ *         "type": "bip122:000000000019d6689c085ae165831e93/slip44:0",
+ *         "unit": "BTC"
+ *       }
+ *     },
+ *     {
+ *       "address": "bc1qrp0yzgkf8rawkuvdlhnjfj2fnjwm0m8727kgah",
```

Comment:
Just to avoid re-using the same address here:

```suggestion
 *       "address": "bc1qwl8399fz829uqvqly9tcatgrgtwp3udnhxfq4k",
```